src/communication/CommandLine.cpp

```diff
@@ -0,0 +1,475 @@
+    //TODO: Do we want to print a progress bar? We'd need a better solution to not have that progress bar be ruined by any logging.
+}
+
+void CommandLine::sliceNext()
+{
+    FffProcessor::getInstance()->time_keeper.restart();
+
+    //Count the number of mesh groups to slice for.
+    size_t num_mesh_groups = 1;
+    for (size_t argument_index = 2; argument_index < arguments.size(); argument_index++)
+    {
+        if (arguments[argument_index].find("--next") == 0) //Starts with "--next".
+        {
+            num_mesh_groups++;
+        }
+    }
+    Slice slice(num_mesh_groups);
+
+    Application::getInstance().current_slice = &slice;
+
+    size_t mesh_group_index = 0;
+    Settings& last_settings = slice.scene.mesh_groups[mesh_group_index].settings;
+
+    slice.scene.extruders.emplace_back(0, &slice.scene.settings); //Always have one extruder.
+    ExtruderTrain& last_extruder = slice.scene.extruders[0];
+
+    for (size_t argument_index = 2; argument_index < arguments.size(); argument_index++)
+    {
+        std::string argument = arguments[argument_index];
+        if (argument[0] == '-') //Starts with "-".
+        {
+            if (argument[1] == '-') //Starts with "--".
+            {
+                if (argument.find("--next") == 0) //Starts with "--next".
+                {
+                    try
+                    {
+                        log("Loaded from disk in %5.3fs\n", FffProcessor::getInstance()->time_keeper.restart());
+
+                        mesh_group_index++;
+                        FffProcessor::getInstance()->time_keeper.restart();
+                        last_settings = slice.scene.mesh_groups[mesh_group_index].settings;
+                    }
+                    catch(...)
+                    {
+                        //Catch all exceptions.
+                        //This prevents the "something went wrong" dialogue on Windows to pop up on a thrown exception.
+                        //Only ClipperLib currently throws exceptions. And only in the case that it makes an internal error.
+                        logError("Unknown exception!\n");
+                        exit(1);
+                    }
+                }
+                else
+                {
+                    logError("Unknown option: %s\n", argument.c_str());
+                }
+            }
+            else //Starts with "-" but not with "--".
+            {
+                argument = arguments[argument_index];
+                switch(argument[1])
+                {
+                    case 'v':
+                        increaseVerboseLevel();
+                        break;
+#ifdef _OPENMP
+                    case 'm':
+                    {
+                        int threads = stoi(argument.substr(1));
+                        threads = std::max(1, threads);
+                        omp_set_num_threads(threads);
+                        break;
+                    }
+#endif //_OPENMP
+                    case 'p':
```

Comment:
Suggest to put brackets for more clarity.

Reply:
Done: c79a886a5dda870f325fff8eaf3c9241ef137f33